### PR TITLE
Fix deprecations for up-to-date chisel/FIRRTL API

### PR DESCRIPTION
**Pull request by sequencer**

### CODE REVIEW

**Comment by davidbiancolin** [comment]
I think switching to stage is increasing JVM memory usage and the heap bound set here:
https://github.com/firesim/firesim/blob/master/.circleci/config.yml#L9

Is too large, so the JVM is culled before GC kicks in.  Could you try reducing that to 3000M?

**Comment by sequencer** [comment]
> Is too large, so the JVM is culled before GC kicks in. Could you try reducing that to 3000M?

I'll have a try. 

but why does FIRRTL have this performance issue, IIRC, dependency API only adds a DAG to manage dependency? Is that true? @seldridge

**Comment by davidbiancolin** [comment]
I don't think this is a performance issue, I think our CI simply over-provisioned heap space. Switching to stage probably pushed a memory-marginal design close to the limit.

**Comment by davidbiancolin** [comment]
@sequencer Can you rebase this? It should just work now.

**Comment by sequencer** [comment]
sorry for the late reply. I’ll work on this tomorrow.

**Comment by sequencer** [comment]
Oh, I totally forgot this, sorry...

**Comment by sequencer** [comment]
```
  Downloading https://files.pythonhosted.org/packages/d8/ba/21c475ead997ee21502d30f76fd93ad8d5858d19a3fad7cd153de698c4dd/bcrypt-3.2.0.tar.gz (42kB)
    Complete output from command python setup.py egg_info:
    Couldn't find index page for 'cffi' (maybe misspelled?)
    No local packages or download links found for cffi>=1.1
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-uZHgyE/bcrypt/setup.py", line 76, in <module>
        cmdclass={"test": PyTest},
      File "/usr/lib64/python2.7/distutils/core.py", line 112, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 265, in __init__
        self.fetch_build_eggs(attrs.pop('setup_requires'))
      File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 289, in fetch_build_eggs
        parse_requirements(requires), installer=self.fetch_build_egg
      File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 618, in resolve
        dist = best[req.key] = env.best_match(req, self, installer)
      File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 862, in best_match
        return self.obtain(req, installer) # try and download/install
      File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 874, in obtain
        return installer(requirement)
      File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 339, in fetch_build_egg
        return cmd.easy_install(req)
      File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 617, in easy_install
        raise DistutilsError(msg)
    distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('cffi>=1.1')
```
CI breaks, but I think that's not my fault.

**Comment by davidbiancolin** [comment]
You rebased this against the dev? Weird, it's still using the old CI.

File: docs/Initial-Setup/Setting-up-your-Manager-Instance.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ To launch a manager instance, follow these steps:
    data is preserved when you stop/start the instance, and your data is
    not lost when pricing spikes on the spot market.
 2. When prompted to select an AMI, search in the ``Community AMIs`` tab for
-   ``FPGA Developer AMI - 1.6.0`` and select the AMI that appears (there 
+   ``FPGA Developer AMI - 1.6.1`` and select the AMI that appears (there
```
Comment (sequencer) [inline docs/Initial-Setup/Setting-up-your-Manager-Instance.rst:28]: Should I remove this?

**Comment by sequencer** [comment]
> You rebased this against the dev branch? Weird, it's still using the old CI.

I think I should reopen this PR?

**Comment by davidbiancolin** [comment]
Try it. I'm going to have to pull in into a local branch for AWS CI to run properly. 

Don't worry about the doc change, that's fine.

**Pull request closed without merging**